controller/src/lib.rs

```diff
@@ -217,6 +217,11 @@ pub enum Error {
     /// A failure interacting with a transceiver module.
     #[error("Error interacting with transceiver module")]
     Transceiver(#[from] TransceiverError),
+
+    /// A failure when an I2C transaction will attempt to access outside the
+    /// 256 memory map size.
+    #[error("Error constructing an I2C transaction")]
```

Comment:
I'd probably make the message something more explicit about the endpoint being out of range. Maybe `End of read (<the value>) is out of the range [0, 255]`?